zcash_client_backend/src/fees/zip317.rs

```diff
@@ -527,7 +598,7 @@ mod tests {
         assert_matches!(
             result,
             Ok(balance) if
-                balance.proposed_change() == [ChangeValue::sapling(Zatoshis::ZERO, None)]
+                balance.proposed_change().is_empty()
```

Comment:
Please explain why this case changes.

Reply:
This was related to the change that I am reverting after your comment on https://github.com/zcash/librustzcash/pull/1734#discussion_r1985418687